### PR TITLE
Add a topic about reaching data that selectors cannot reach

### PR DESCRIPTION
**Pull request by Gallaecio**
Write a new topic that addresses [Can scrapy be used to scrape dynamic content from websites that are using AJAX?](https://stackoverflow.com/q/8550114/939364), one of the StackOverflow questions about Scrapy with the most votes.

Note to self: https://docs.scrapy.org/en/latest/topics/developer-tools.html

### CODE REVIEW

**Comment by dangra** [comment]
hi @Gallaecio, this doc topic looks good and ready to merge, isn't it?

**Comment by Gallaecio** [comment]
It is ready by me. But maybe someone else (@kmike?) wants to review it first.

File: docs/topics/beyond-selectors.rst [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+request with Scrapy.
+
+It might be enough to yield a :class:`~scrapy.http.Request` with the same HTTP
+method and URL. However, you may also need to reproduce the body, headers and
+form parameters (see :class:`~scrapy.http.FormRequest`) of that request.
+
+Once you get the expected response, you can :ref:`extract the desired data from
+it <topics-handling-response-formats>`.
+
+If you cannot manage to get the expected response,
```
Comment (Gallaecio) [inline docs/topics/beyond-selectors.rst]: Actually, it seems I did not finish writing here :flushed:

**Comment by Gallaecio** [comment]
Completed.

**Comment by kmike** [comment]
Hi @Gallaecio! I think it is a much-needed section of the documentation. I've checked it, and it looks good to me, not comments or suggestions :) 

However, I wonder if that's possible to find a better name for this section. "Data which selectors can't reach", "beyond selectors" doesn't sound precise - the problem is not with selectors (and in fact we're suggesting to use them later), the problem is that data is not available in the downloaded HTML, or that data can't be easily extracted by parsing HTML itself, and instead requires parsing some other format embedded in HTML (such as CSS or JS).

**Comment by Gallaecio** [comment]
I must say I’m not happy with the name either. Any suggestions?

**Comment by Gallaecio** [comment]
If we extend the topic to cover the easiest scenario as well, that where selectors work as expected, we could go for a simple, generic title in the lines of “Extracting data” or “Locating and extracting the desired data”.

**Comment by raphapassini** [comment]
I would vote for something in the lines "Selecting dynamically loaded content"

**Comment by Gallaecio** [comment]
Hmm… "Dynamically-loaded" definitely fits everything covered one way or another. I like it.

**Comment by kmike** [comment]
👍 👍

**Comment by pablohoffman** [comment]
great addition to the doc! 💪

**Pull request merged**